src/util/SweptCollection.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const Collection = require('./Collection.js');
+const { TypeError } = require('../errors/DJSError.js');
+
+/**
+ * @typedef {Function} SweepFilter
+ * @param {SweptCollection} collection The collection being swept
+ * @returns {Function|false} Return `false` to skip sweeping, otherwise a function passed to `sweep()`,
+ * See {@link [Collection#sweep](https://discord.js.org/#/docs/collection/master/class/Collection?scrollTo=sweep)}
+ * for the definition of this function.
+ */
+
+/**
+ * Options for defining the behavior of a Swept Collection
+ * @typedef {Object} SweptCollectionOptions
+ * @property {Function|null} [sweepFilter=null] A function run every `sweepInterval` to determine how to sweep
+ * @property {number} [sweepInterval=3600] How frequently, in seconds, to sweep the collection.
+ */
+
+/**
+ * A Collection which holds a max amount of entries and sweeps periodically.
+ * @extends {Collection}
+ * @param {SweptCollectionOptions} [options={}] Options for constructing the swept collection.
+ * @param {Iterable} [iterable=null] Optional entries passed to the Map constructor.
+ */
+class SweptCollection extends Collection {
+  constructor(options = {}, iterable) {
+    if (typeof options !== 'object' || options === null) {
+      throw new TypeError('INVALID_TYPE', 'options', 'object or iterable', true);
+    }
+    const { sweepFilter = null, sweepInterval = 3600 } = options;
+
+    if (sweepFilter !== null && typeof sweepFilter !== 'function') {
+      throw new TypeError('INVALID_TYPE', 'sweepFunction', 'function');
+    }
+    if (typeof sweepInterval !== 'number') throw new TypeError('INVALID_TYPE', 'sweepInterval', 'number');
+
+    super(iterable);
+
+    /**
+     * A function called every sweep interval that returns a function passed to `sweep`
+     * @type {?Function}
+     */
+    this.sweepFilter = sweepFilter;
+
+    /**
+     * The id of the interval being used to sweep.
+     * @type {?Number}
```

Comment:
```suggestion
     * @type {?number}
```
why just the id? why not the interval object?

Reply:
it *is* the object, I just still associate the return value with the id 😆